ambry-router/src/main/java/com/github/ambry/router/CompressionMetrics.java

```diff
@@ -0,0 +1,141 @@
+  public final Counter compressSkip_ContentEncoding;
+  public final Counter compressSkip_ContentTypeFiltering;
+  public final Counter compressSkip_SizeTooSmall;
+  public final Counter compressSkip_RatioTooSmall;
+
+  // Decompression metrics, regardless of decompression algorithms.
+  public final Meter decompressSuccessRate;
+  public final Meter decompressErrorRate;
+  public final Counter decompressExpandSizeBytes;  // Bytes expand  = OriginalSize - CompressedSize
+  public final Counter decompressError_BufferConversion;
+  public final Counter decompressError_BufferTooSmall;
+  public final Counter decompressError_UnknownAlgorithmName;
+  public final Counter decompressError_DecompressFailed;
+
+  /**
+   * Create an instance of compression metrics.
+   *
+   * @param registry The metrics registry to register its metrics.
+   */
+  public CompressionMetrics(MetricRegistry registry) {
+    metricsRegistry = registry;
+
+    // Compression metrics.
+    compressSkipRate = registry.meter(MetricRegistry.name(ownerClass,"CompressSkipRate"));
+    compressErrorRate = registry.meter(MetricRegistry.name(ownerClass,"CompressErrorRate"));
+    compressAcceptRate = registry.meter(MetricRegistry.name(ownerClass,"CompressAcceptRate"));
+    compressReduceSizeBytes = registry.counter(MetricRegistry.name(ownerClass, "CompressReduceSizeBytes"));
+    compressError_BufferConversion = registry.counter(MetricRegistry.name(ownerClass,"CompressErrorBufferConversion"));
+    compressError_ConfigInvalidCompressorName = registry.counter(MetricRegistry.name(ownerClass,"CompressErrorInvalidCompressorName"));
+    compressError_CompressFailed = registry.counter(MetricRegistry.name(ownerClass,"CompressErrorCompressFailed"));
+    compressSkip_ContentEncoding = registry.counter(MetricRegistry.name(ownerClass,"CompressSkipContentEncoding"));
+    compressSkip_ContentTypeFiltering = registry.counter(MetricRegistry.name(ownerClass,"CompressSkipContentTypeFiltering"));
+    compressSkip_SizeTooSmall = registry.counter(MetricRegistry.name(ownerClass,"CompressSkipSizeTooSmall"));
+    compressSkip_RatioTooSmall = registry.counter(MetricRegistry.name(ownerClass,"CompressSkipRatioTooSmall"));
+
+    // Decompression metrics.
+    decompressSuccessRate = registry.meter(MetricRegistry.name(ownerClass,"DecompressSuccessRate"));
+    decompressErrorRate = registry.meter(MetricRegistry.name(ownerClass,"DecompressErrorRate"));
+    decompressExpandSizeBytes = registry.counter(MetricRegistry.name(ownerClass,"DecompressExpandSizeBytes"));
+    decompressError_BufferConversion = registry.counter(MetricRegistry.name(ownerClass,
+        "DecompressErrorBufferConversion"));
+    decompressError_BufferTooSmall = registry.counter(MetricRegistry.name(ownerClass,
+        "DecompressErrorBufferTooSmall"));
+    decompressError_UnknownAlgorithmName = registry.counter(MetricRegistry.name(ownerClass,
+        "DecompressErrorUnknownAlgorithmName"));
```

Comment:
will this ever happen ? if this happens, then it means we can't read the blob ?

Reply:
This shouldn't happen.  If they do, we need to investigate immediately because it could mean data lost.  That's why we need to monitor these metrics and setup alarms.